src/main/java/frc/robot/leds/LedState.java

```diff
@@ -0,0 +1,59 @@
+package frc.robot.leds;
+
+import java.awt.*;
+
+public class LedState {
+    private LedMode mode;
+    private Color primary;
+    private Color secondary;
+    private double blinkTime;
+    private double fadeDuration;
+
+    public LedState(LedMode mode, Color primary, Color secondary, double blinkTime, double fadeDuration) {
```

Comment:
Perhaps a `record` will fit here more than a regular class? A `record` is immutable, but you could just create a few `LedState`s and switch between them.